### PR TITLE
Create a UserAvatar component

### LINKED ISSUE #2857: (SP: 3) Create UserAvatar component aligned with design system
#### Description:
Implement a reusable `UserAvatar` component that meets the design system standards. This component will be used throughout the application to maintain consistency in user avatar styles, behavior, and accessibility.

[[Figma](https://www.figma.com/design/liJZDYhUnDP15Pi9bipDcv/Space2Study-(Students)?node-id=21376-87013&t=l5PI8JRXpNFKQBr0-4)]

#### Requirements:

**Styles**:

Follow design system variables for colors, icons, padding, and other styles.
-Support different variants (check, monogram, avatar, photo).
-Provides a label that signals online status

**Props**:

-`variant`: Defines type (check, monogram, avatar, photo).
-`size`: Supports different sizes (sm, md, lg).
-`src`: Provide URL for photo (for the type photo)
-`monogram`: Provide text for the monogram 
-`isOnline`: The label is displayed when true(user is online).
-`onClick`: A callback function that processes the event of clicking on the icon.

Don't forget to cover component with tests and add to the storybook

### PR DESCRIPTION
**Pull request by PavloButynets**
Implemented a reusable UserAvatar component that meets the design system standards. Covered the tests component and added to the storybook
image
image
image

### CODE REVIEW

File: tests/unit/design-system/components/user-avatar/UserAvatar.spec.jsx
Comment (amoutens) [inline tests/unit/design-system/components/user-avatar/UserAvatar.spec.jsx:1]: please fix Sonar Issue and add more coverage on tests

**Review by amoutens — CHANGES_REQUESTED** [review]

File: src/design-system/components/user-avatar/UserAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            alt={monogram}
+            className={cn('s2s-avatar', `s2s-avatar-${size}`)}
+            onClick={onClick}
+            ref={forwardedRef}
+            src={src}
+            {...props}
+          />
+        ) : variant === 'monogram' ? (
+          <MuiAvatar
+            onClick={onClick}
```
Comment (amoutens) [inline src/design-system/components/user-avatar/UserAvatar.tsx]: you can replace `cn('s2s-avatar', `s2s-avatar-${size}`)` to variable

File: src/design-system/components/user-avatar/UserAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            ref={forwardedRef}
+            {...props}
+            className={cn('s2s-avatar', `s2s-avatar-${size}`)}
+          />
+        )}
+        {isOnline && (
+          <span
+            className={`s2s-user-avatar-status s2s-user-avatar-status-${size}`}
```
Comment (amoutens) [inline src/design-system/components/user-avatar/UserAvatar.tsx]: use cn for this className

**Review by amoutens — CHANGES_REQUESTED** [review]

File: src/design-system/components/user-avatar/UserAvatar.tsx (hunk 1 above)
Comment (PavloButynets) [reply src/design-system/components/user-avatar/UserAvatar.tsx:56]: For what? This is not in our rules for code

File: src/design-system/components/user-avatar/UserAvatar.tsx (hunk 2 above)
Comment (PavloButynets) [reply src/design-system/components/user-avatar/UserAvatar.tsx:82]: for 1 className it isn't necessary

**Review by amoutens — APPROVED** [review]

File: src/design-system/components/user-avatar/UserAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      src,
+      size = 'sm',
+      firstName,
+      lastName,
+      isOnline,
+      onClick,
+      ...props
+    }: UserAvatarProps,
+    forwardedRef: Ref
+  ) => {
```
Comment (Made1ra) [inline src/design-system/components/user-avatar/UserAvatar.tsx:39]: It is better to use generic in `forwardRef` function. You will not have to type props and forwardedRef manually.
```suggestion
const UserAvatar = forwardRef<HTMLDivElement, UserAvatarProps>(
  (
    {
      variant = 'avatar',
      src,
      size = 'sm',
      firstName,
      lastName,
      isOnline,
      onClick,
      ...props
    },
   ref
  ) => {
```

Comment (PavloButynets) [reply src/design-system/components/user-avatar/UserAvatar.tsx:39]: image
the DOM element type is HTMLDivElement at all levels, so I don't see the point in specifically specifying HTMLDivElement for ref

**Review by RomanRozbytskyi — APPROVED** [review]

**Review by SofiiaYevush — APPROVED** [review]

File: src/design-system/components/user-avatar/UserAvatar.tsx (hunk 3 above)
Comment (Made1ra) [reply src/design-system/components/user-avatar/UserAvatar.tsx:39]: You can keep the ref, but it's better to extract it into a generic `forwardRef<Ref, UserAvatarProps>`.

**Pull request merged**